tests/ConnectionTests.ts

```diff
@@ -913,3 +917,54 @@ test("Test InjectMessage", (done: jest.DoneCallback) => {
         }, done.fail);
     }, done.fail);
 });
+
+describe("Connection errors are retried", () => {
+    let errorCount: number;
+    let detachObject: IDetachable;
+
+    beforeEach(() => {
+        errorCount = 0;
+        detachObject = Events.instance.attachListener({
+            onEvent: (event: PlatformEvent) => {
+                if (event instanceof ConnectionErrorEvent) {
+                    const connectionEvent: ConnectionErrorEvent = event as ConnectionErrorEvent;
+                    errorCount++;;
+                }
+            },
+        });
+    });
+
+    afterEach(() => {
+        if (undefined !== detachObject) {
+            detachObject.detach().catch((error: string) => {
+                throw new Error(error);
+            });
+            detachObject = undefined;
+        }
+    });
+
+    test.only("Bad Auth", (done: jest.DoneCallback) => {
+        const s: sdk.SpeechConfig = sdk.SpeechConfig.fromSubscription("badKey", Settings.SpeechRegion);
+        const ps: sdk.PushAudioInputStream = sdk.AudioInputStream.createPushStream();
+
+        const r: sdk.SpeechRecognizer = new sdk.SpeechRecognizer(s, sdk.AudioConfig.fromStreamInput(ps));
+        objsToClose.push(r);
+
+        r.recognizeOnceAsync((result: sdk.SpeechRecognitionResult) => {
+            try {
+                expect(sdk.ResultReason[result.reason]).toEqual(sdk.ResultReason[sdk.ResultReason.Canceled]);
+                const canceledDetails: sdk.CancellationDetails = sdk.CancellationDetails.fromResult(result);
+                expect(sdk.CancellationReason[canceledDetails.reason]).toEqual(sdk.CancellationReason[sdk.CancellationReason.Error]);
+                expect(sdk.CancellationErrorCode[sdk.CancellationErrorCode.ConnectionFailure]).toEqual(sdk.CancellationErrorCode[canceledDetails.ErrorCode]);
+                expect(errorCount).toEqual(5);
```

Comment:
Any reason not to set the retry property on the config and then check against that?